crates/apollo-compiler/src/validation/field.rs

```diff
@@ -12,7 +12,67 @@ pub fn validate_field(
 ) -> Vec<ApolloDiagnostic> {
     let mut diagnostics =
         db.validate_directives(field.directives().to_vec(), hir::DirectiveLocation::Field);
-    diagnostics.extend(db.validate_arguments(field.arguments().to_vec()));
+
+    if !field.arguments().is_empty() {
+        diagnostics.extend(db.validate_arguments(field.arguments().to_vec()));
+    }
+
+    if let Some(field_definition) = field.field_definition(db.upcast()) {
+        for arg in field.arguments() {
+            let exists = field_definition
+                .arguments()
+                .input_values()
+                .iter()
+                .any(|arg_def| arg.name() == arg_def.name());
+
+            if !exists {
+                let diagnostic = ApolloDiagnostic::new(
+                    db,
+                    arg.loc.into(),
+                    DiagnosticData::UndefinedArgument {
+                        name: arg.name().into(),
+                    },
+                )
+                .label(Label::new(arg.loc, "argument by this name not found"))
+                .label(Label::new(field_definition.loc, "field declared here"));
+
+                diagnostics.push(diagnostic);
+            }
+        }
```

Comment:
should we add this to `validate_arguments` instead? i feel like that could be called from both `field` and `directive`.

Reply:
Ah yes. i started with that but to build the diagnostics, we have to refer to different types (FieldDefinition and DirectiveDefinition respectively), so it's not trivial. I think duplicating the code is easier than trying to introduce some abstraction for it or trying to share more functionality.